llvm/test/Transforms/InstCombine/powi.ll

```diff
@@ -258,3 +259,23 @@ define double @different_types_powi(double %x, i32 %y, i64 %z) {
   %mul = fmul reassoc double %p2, %p1
   ret double %mul
 }
+
+define nofpclass(nan inf) double @fdiv_pow_powi(double %x) {
+; CHECK-LABEL: @fdiv_pow_powi(
+; CHECK-NEXT:    [[DIV:%.*]] = fmul fast double [[X:%.*]], [[X]]
+; CHECK-NEXT:    ret double [[DIV]]
+;
+  %p1 = call fast double @llvm.powi.f64.i32(double %x, i32 3)
+  %div = fdiv fast double %p1, %x
```

Comment:
Please check the precise fmf flags here (and also test that the fold is not performed if one of the required flags is missing).

Reply:
hi @nikic,
  I find that **I.getFastMathFlags().isFast()** is used directly in very few scenarios. Is it too strict?